### ISSUE TITLE
Problem with how some metrics are averaged (not intuitive)

### ISSUE THREAD

**Issue by znb899** [Lightning-AI/torchmetrics#2094]
## 🐛 Bug

It seems to me the precision is calculated (when global averaging), be it with update or compute, as:

(TP1 + TP2 + TP3....) / ((TP1 + TP2 + TP3....) + (FP1 + FP2 + FP3....) )

When it makes more sense to have (or at leat have the option):

mean([TP1 / (TP1 + FP1)] + [TP2 / (TP2 + FP2)] + [TP3 / (TP3 + FP3)]...)

Which is not the same

The same goes for f-score, recall but the IoU. So whatever the reason to do it the first way, it's not consistent

**Comment by SkafteNicki** [comment]
Hi @znb899, thanks for raising this issue.
What you are describing is essentially the difference between `micro` and `macro` averaging. Doing the calculation like this
```
(TP1 + TP2 + TP3....) / ((TP1 + TP2 + TP3....) + (FP1 + FP2 + FP3....) )
```
corresponds to `micro` averaging and doing it like this
```
mean([TP1 / (TP1 + FP1)] + [TP2 / (TP2 + FP2)] + [TP3 / (TP3 + FP3)]...)
```
corresponds to `macro` averaging. There is no right or wrong way to do the averaging each of the two have pros and cons. They can especially be very different when you have large class imbalance. [Here](https://datascience.stackexchange.com/questions/15989/micro-average-vs-macro-average-performance-in-a-multiclass-classification-settin) is a good discussion about the difference between them.
All that said, Torchmetrics absolutely support calculating metrics like precision, recall, f-score etc using both types of averaging:
```python
from torchmetrics.functional.classification import multiclass_precision
import torch

preds = torch.randn(100, 3).softmax(dim=1)
target = torch.randint(0, 3, (100,))

print(multiclass_precision(preds, target, num_classes=3, average='macro'))
print(multiclass_precision(preds, target, num_classes=3, average='micro'))
```

**Comment by znb899** [comment]
Hi @SkafteNicki Thank for your answer
You're right!

**Comment by znb899** [comment]
I think there a confusion between averaging of labels and averaging of batches. For examples, how do I get the precision of all the class where it's the avarage of the precisions on every samples?
